adoc/chapters/programming_interface.adoc

```diff
@@ -2401,8 +2401,6 @@ info::device::driver_version
 
     @ [.code]#std::string#
    a@ Returns a backend-defined driver version as a [code]#std::string#.
-      If using the OpenCL backend, the returned value represents the
-      OpenCL software driver version in the form: major_number.minor_number.
```

Comment:
I see that the OpenCL backend interop spec says this corresponds to `CL_DRIVER_VERSION`.  OpenCL 3.0 defines that as:

> OpenCL software driver version string. Follows a vendor-specific format.

Since OpenCL doesn't mandate any specific format, should we just say this in the SYCL spec directly?  The SYCL definition would then be something like:

> A vendor-defined string describing the version of the underlying backend software driver.

Reply:
Oh it got updated in 3.0! That's a better definition for sure.

> A vendor-defined string describing the version of the underlying backend software driver.

I feel like this a bit in conflict with the idea of having it backend-defined, the way I see it we say it's backend-defined in the main spec, then in the backend spec we can define it further to vendor-defined if we want, for the OpenCL plugin saying it's `CL_DRIVER_VERSION` defers it to the OpenCL spec which ends up being vendor defined anyway.

But then for say a CUDA backend we could define it to the compute capability in the format `major.minor` for example rather than having it vendor defined.

But if we just say it's vendor defined in the main spec I feel like we can't really give more details on what it's supposed to be in the backend spec.

Reply:
What is the motivation for specifying this query as backend-defined, though?  Most of our queries are not backend-defined.  If the intent of `driver_version` is to return the version string of the vendor's device driver, it's hard to see why a backend would mandate any specific format.  Wouldn't we want each vendor to have the freedom to decide it's own format?

Reply:
Oh right! I was still thinking about the `version` rather than the `driver_version`, since the `driver_version` was already backend specific before my patch.

But now that I think about it, I do think you're right that it makes more sense to have the `driver_version` simply be vendor specific, I'll change it as well.

Reply:
Updated the `driver_version` as suggested and removed the `CL_DRIVER_VERSION` part from the OpenCL backend spec.